tck/src/main/java/ee/jakarta/tck/json/bind/api/jsonb/JsonbTest.java

```diff
@@ -252,10 +261,135 @@ public void testToJsonObjectStream() throws IOException {
     @Test
     public void testToJsonObjectTypeStream() throws IOException {
         try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
-            jsonb.toJson(new SimpleContainer(), new SimpleContainer() { }.getClass().getGenericSuperclass(), stream);
-            String jsonString = new String(stream.toByteArray(), StandardCharsets.UTF_8);
+            CloseRememberingOutputStream rememberingStream = new CloseRememberingOutputStream(stream);
+            jsonb.toJson(new SimpleContainer(), new SimpleContainer() { }.getClass().getGenericSuperclass(), rememberingStream);
+            String jsonString = new String(rememberingStream.toByteArray(), StandardCharsets.UTF_8);
             assertThat("Failed to marshal using Jsonb.toJson method with Object, Type and OutputStream arguments.",
                        jsonString, matchesPattern(MATCHING_PATTERN));
+            assertThat("Failed to close stream upon a successful completion", rememberingStream.isCloseCalled());
+        }
+    }
+
+    static final class CloseRememberingOutputStream extends OutputStream {
+
+        private ByteArrayOutputStream delegate;
+        private boolean closeCalled;
+
+        CloseRememberingOutputStream(ByteArrayOutputStream delegate) {
+            this.delegate = delegate;
+            this.closeCalled = false;
+        }
+
+        @Override
+        public void close() throws IOException {
+            closeCalled = true;
+            delegate.close();
+        }
+
+        boolean isCloseCalled() {
+            return closeCalled;
+        }
+
+        byte[] toByteArray() {
```

Comment:
missing `@Override`

Reply:
Can't do that `#toByteArray()` is defined on `ByteArrayOutputStream` not on `OutputStream`.